mongo/description/server.go

```diff
@@ -48,7 +48,7 @@ type Server struct {
 	SessionTimeoutMinutes   uint32
 	SetName                 string
 	SetVersion              uint32
-	SpeculativeAuthenticate bsoncore.Document
+	SpeculativeAuthenticate bson.Raw
```

Comment:
Should SpeculativeAuthenticate be private?

Authentication does not occur on monitoring connections: https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-monitoring.rst#servers-are-monitored-with-dedicated-sockets

> Drivers MUST NOT authenticate on sockets used for monitoring nor include SCRAM mechanism negotiation (i.e. saslSupportedMechs), as doing so would make monitoring checks more expensive for the server.

So speculative authentication only occurs on application connections during the initial handshake, but not on monitoring connections. So I believe this would always be empty for SDAM monitoring events.

(FWIW, since libmongoc has some single-threaded weirdness where it uses the same socket for monitoring and application use, we had to scrub the speculative authentication payload out of the SDAM monitoring events. Like [auth commands](https://github.com/mongodb/specifications/blob/master/source/command-monitoring/command-monitoring.rst#security), they should be redacted from monitoring events).

Reply:
will be handled in GODRIVER-1740